### PR TITLE
Acoustic comms refactoring

### PR DESCRIPTION
**Pull request by adityapande-1995**
This PR attempts to use the upstream acoustic comms plugin (https://github.com/gazebosim/gz-sim/pull/1608) in gz-sim, and replace the implementation here.

### CODE REVIEW

**Comment by adityapande-1995** [comment]
The failing tests are also failing on ``main``. The only thing left here is removing the old ``AcousticComms.h/cc`` files and its remnants from ``CMakeLists.txt``

**Review by arjo129 — COMMENTED** [review]
Seems OK to me. We just need to make the tests pass.

File: lrauv_system_tests/test/simulation/test_spawn.cc [hunk 1]
```diff
@@ -149,16 +149,16 @@ TEST(VehicleSpawnTest, Spawn)
...
-      topics.begin(), topics.end(),
-      [](const std::string &s) {
-        return s == "/comms/external/201/tx";
-      }));
-  EXPECT_TRUE(std::any_of(
-      topics.begin(), topics.end(),
-      [](const std::string &s) {
-        return s == "/comms/external/202/tx";
-      }));
+  /* EXPECT_TRUE(std::any_of( */
```
[the remark this answers is not in the document]
Comment (adityapande-1995) [reply lrauv_system_tests/test/simulation/test_spawn.cc]: Done

**Comment by adityapande-1995** [comment]
Investigating test failures

**Comment by adityapande-1995** [comment]
@arjo129 we have CI passing here, after review I can remove the old AcousticComms.cc and .hh files.

**Comment by adityapande-1995** [comment]
@arjo129 , we have CI passing here, and I can remove the old AcousticComms.cc and hh files after review.

**Review by hidmic — APPROVED** [review]

**Pull request merged**